### ISSUE TITLE
I update an item that does not exit in table, but return yes.

### ISSUE THREAD

**Issue by BarryGuo** [ccgus/fmdb#471]
Not found a "target" that equal to "1923424344",but when i run this code ，return yes。
![d2c2341f-a82e-4bab-9ba4-57454835cff6](https://cloud.githubusercontent.com/assets/16474028/13568258/044affc2-e49b-11e5-84ab-8f8ae1c62ed3.png)

**Comment by ccgus** [comment]
Can you explain more what the problem is?

**Comment by nshetty26** [comment]
I believe I am also having this problem.  There are rows that I know should be in the DB, followed through code and gotten the successful return on the executeUpdate, but when I look at the DB file the row is not there.

Some information:
1. Using FMDB in Swift from CocoaPods using Frameworks
2. I am running FMDB with Sql Cipher
3. I am running all my inserts through an NSOperation which is processed in a serial queue
4. I am running every transaction in an FMDBQueue 
4. There maybe a correlation between amount of writes and writes being lost.

If you can give me debugging tips on where I should be looking I would be happy to dive deeper.

**Comment by Allendar** [comment]
Update is always successful when there are not errors. What you're likely looking for is the `sqlite3_changes` ([_Core.yourDB changes]).

After this you're still not fully done if you really want specific feedback, because update is still triggered as a change when it's internal cursor finds a match.

Image this simple query:

    UPDATE `settings` SET `value` = '1' WHERE `key` = 'show_matrix_by_default'

This will always say it has one change, no matter what `value` was before. So even if `value` was already 1 it will still hit for an update.

If you really want specifics (this will be more costly without the correct indexes tho) is:

    UPDATE `settings` SET `value` = '1' WHERE `key` = 'show_matrix_by_default' AND `value` = '0'

Because you dictate your cursor's parameters to be more specific it will only show 1 change if value was `0` (OFF) and swap it to `1` (ON).

Of course this only works for specific row-hits. For multiple hits and more complex lookups you probably need more code or variations of `WHERE IN`, `JOIN`, `UNION`, etc.

**Comment by robertmryan** [comment]
Yes, SQL with `WHERE` clause will not return an error if there were no rows affected. If only returns an error if there was a syntax error in the SQL. If you have `INSERT`, `UPDATE` or `DELETE` statement with `WHERE` clause, use `changes` to confirm how many rows were affected.
